src/platform/OpenThread/GenericNetworkCommissioningThreadDriver.cpp

```diff
@@ -47,6 +47,7 @@ CHIP_ERROR GenericThreadDriver::Init(Internal::BaseDriver::NetworkStatusChangeCa
 {
     ThreadStackMgrImpl().SetNetworkStatusChangeCallback(statusChangeCallback);
     ThreadStackMgrImpl().GetThreadProvision(mStagingNetwork);
+    PlatformMgr().AddEventHandler(OnThreadStateChangeHandler, reinterpret_cast<intptr_t>(&mStagingNetwork));
```

Comment:
```suggestion
    ReturnErrorOnFailure(PlatformMgr().AddEventHandler(OnThreadStateChangeHandler, reinterpret_cast<intptr_t>(&mStagingNetwork)));
```
Also, you could probably pass `this` instead of a single member to have the handler more extensible if needed in the future.

Reply:
okay, have been modified to pass `this` as the parameter.

Reply:
Thanks!